dogowner/test_dog_owner.py

```diff
@@ -0,0 +1,97 @@
+                            phone_number=PHONE_NUMBER, dog_race=DOG_RACE, dog_picture_url=DOG_PICTURE_URL,
+                            dog_age=DOG_AGE, dog_weight=DOG_WEIGHT, dog_gender=DOG_GENDER
+                            )
+
+    def test_same_username(self, dogOwner_01):
+        dogOwner_01.save()
+        with pytest.raises(ValidationError, match="Invalid username"):
+            DogOwner.create(email=EMAIL, username='testuser01', password=PASSWORD,
+                            dog_name=DOG_NAME, first_name=FIRST_NAME, last_name=LAST_NAME,
+                            phone_number=PHONE_NUMBER, dog_race=DOG_RACE, dog_picture_url=DOG_PICTURE_URL,
+                            dog_age=DOG_AGE, dog_weight=DOG_WEIGHT, dog_gender=DOG_GENDER
+                            )
+
+    def test_invalid_url(self, dogOwner_01):
+        with pytest.raises(ValidationError, match="Enter a valid URL."):
+            DogOwner.create(email=EMAIL, username='testuser03', password=PASSWORD,
+                            dog_name=DOG_NAME, first_name=FIRST_NAME, last_name=LAST_NAME,
+                            phone_number=PHONE_NUMBER, dog_race=DOG_RACE, dog_picture_url='NOT_A_URL',
+                            dog_age=DOG_AGE, dog_weight=DOG_WEIGHT, dog_gender=DOG_GENDER
+                            )
+
+    def test_phone_as_txt(self, dogOwner_01):
+        with pytest.raises(ValidationError, match="Invalid phone - phone should be number"):
+            DogOwner.create(email=EMAIL, username='invalid_phone', password=PASSWORD,
+                            dog_name=DOG_NAME, first_name=FIRST_NAME, last_name=LAST_NAME,
+                            phone_number='tests', dog_race=DOG_RACE, dog_picture_url=DOG_PICTURE_URL,
+                            dog_age=DOG_AGE, dog_weight=DOG_WEIGHT, dog_gender=DOG_GENDER
+                            )
+
+    def test_phone_short(self, dogOwner_01):
+        with pytest.raises(ValidationError, match="Invalid phone - phone should be 10 digits"):
+            DogOwner.create(email=EMAIL, username='invalid_phone', password=PASSWORD,
+                            dog_name=DOG_NAME, first_name=FIRST_NAME, last_name=LAST_NAME,
+                            phone_number=123456789, dog_race=DOG_RACE, dog_picture_url=DOG_PICTURE_URL,
+                            dog_age=DOG_AGE, dog_weight=DOG_WEIGHT, dog_gender=DOG_GENDER
+                            )
+
+    def test_phone_long(self, dogOwner_01):
+        with pytest.raises(ValidationError, match="Invalid phone - phone should be 10 digits"):
+            DogOwner.create(email=EMAIL, username='invalid_phone', password=PASSWORD,
+                            dog_name=DOG_NAME, first_name=FIRST_NAME, last_name=LAST_NAME,
+                            phone_number=1234567890123, dog_race=DOG_RACE, dog_picture_url=DOG_PICTURE_URL,
+                            dog_age=DOG_AGE, dog_weight=DOG_WEIGHT, dog_gender=DOG_GENDER
+                            )
```

Comment:
Why is it necessary to call dogOwner_01 fixture?

Reply:
You right , fixed .